File: cpp/14.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <iterator>
#include <utility>
#include <map>
#include <string>
#include <unordered_map>
#include <climits>
using namespace std;

template<typename T>
void prints(vector<T> &v){
  cout << "[";
  for(uint i=0; i<v.size()-1; ++i)
    cout << v[i] << ", ";
  if(v.size()) cout << v[v.size()-1];
  cout << "]\n";
}

class Solution {
public:
  string longestCommonPrefix(vector<string> &strs) {
    int n = strs.size();
    if(!n) return "";
    string ans(strs[0]);
    for(const auto& s : strs){
      auto a = ans.begin();
      auto b = s.begin();
      while(a != ans.end() && b != s.end() && *a == *b) ++a, ++b;
      ans = ans.substr(0, a - ans.begin());
    }
    return ans;
  }
};

int main(){
  auto sol = Solution();
  vector<string> v = {""};
  cout << sol.longestCommonPrefix(v);
}

File: python/14.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ''
        ln = len(strs[0])
        for s in strs[1:]:
            i = 0
            while i < len(s) and i < ln and s[i] == strs[0][i]:
                i += 1
            ln = min(i, ln)
        return strs[0][:ln]
